### DISCUSSION TITLE
Change capabilities of specific server clangd

### DISCUSSION THREAD

**Discussion by serranomorante** [VonHeikemen/lsp-zero.nvim#298] [Q&A] [answered]
Hi, executing lua `require('lsp-zero.check').inspect_server_config('clangd')` throws this result:

```json
...
    offsetEncoding = {"utf-8", "utf-16"}
...
```

How can I update the capability so it only shows `offsetEncoding = {"utf-16"}`?

The following works but **I want to change it for the clangd server only and not for all the servers**

```lua
			lsp.set_server_config({
                                ...
				capabilities = {
					offsetEncoding = "utf-16",
				},
			})

```

This is how I'm configuring my clang server right now, I'm using lsp-zero dev-3

```lua
			require("mason-lspconfig").setup({
				ensure_installed = {
					"clangd",
				},
				handlers = {
					lsp.default_setup,
					clangd = function()
						require("clangd_extensions").setup()
					end,
				},
			})


```

I couldn't find any guide on how to modify the capabilities of a specific lsp server with lsp-zero.

**Comment by VonHeikemen** [comment, marked as answer]
To configure a server (in most cases) you use `lspconfig` directly. Something like this.

```lua
require('mason-lspconfig').setup({
  handlers = {
    lsp.default_setup,
    some_lsp = function()
      require('lspconfig').some_lsp.setup({
        ---
        -- here you can place your
        -- custom config for some_lsp
        ---
      })
    end,
  }
})
```

In this particular case with clangd you have an extra plugin, `clangd_extensions`. Last time I checked, you need to pass the config using a property called `server`.

```lua
require('mason-lspconfig').setup({
  handlers = {
    lsp.default_setup,
    clangd = function()
      require('clangd_extensions').setup({
        server = {
          ---
          -- here you can place your
          -- custom config for clangd
          ---
        }
      })
    end,
  }
})
```

**Reply by serranomorante** [reply]
:sob: :sob: thank you so much, that worked!!

Quick question though, by doing this:

```lua
handlers = {
	lsp.default_setup,
	clangd = function()
		require("clangd_extensions").setup({
			server = {
				capabilities = {
					offsetEncoding = { "utf-16" },
				},
			},
		})
	end,

```

Am I overriding the capabilities provided by `lsp_defaults.capabilities` and `require('cmp_nvim_lsp').default_capabilities()`?

**Reply by VonHeikemen** [reply]
`offsetEncoding` is not configured by cmp_nvim_lsp. You are good.
